plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -388,7 +392,8 @@ private static IcebergSplit toIcebergSplit(FileScanTask task)
                 PartitionData.toJson(task.file().partition()),
                 task.deletes().stream()
                         .map(TrinoDeleteFile::copyOf)
-                        .collect(toImmutableList()));
+                        .collect(toImmutableList()),
+                SplitWeight.fromProportion(Math.max((double) task.length() / tableScan.targetSplitSize(), minimumAssignedSplitWeight)));
```

Comment:
Could `task.length() / tableScan.targetSplitSize()` yield a value > `1.0`? If so, you'll probably want to wrap that with `Math.min(1.0, (double) task.length() / tableScan.targetSplitSize())`

Reply:
Looks like if the file format defines split offsets those are used instead, and I guess they could be bigger than the target split size.

So if a split is larger than standard, we want to cap that at 1.0 rather than accounting for the larger size in the queue?

Reply:
That was the choice I went with in the case of `trino-hive` since I didn't want to regress queries. At some point, you do have to set a cap on the upper bound or you could have splits that never get scheduled because there's never enough room in a worker queue to accept it- but you could choose to go maybe as high as `2.0` if you think that's worthwhile, but the safer bet is definitely `1.0`. I don't know enough about how `tableScan.targetSize()` and split generation in general work in Iceberg to know whether it's a real concern.

Reply:
>  larger weights are allowed if the connector wishes to express that fewer splits should be allowed to be placed per node and task. 

I think connector shouldn't make such a decision. After all, it doesn't control the nodes and the task scheduler and it doesn't know queue lengths, etc.

in fact, this discussion shows that this is (a) an expertly to tune and (b) tuning this is a problem that blurs engine/connector separation. Things may need to be configured on both sides for optimal behavior.

Reply:
> I think connector shouldn't make such a decision. After all, it doesn't control the nodes and the task scheduler and it doesn't know queue lengths, etc.

The connector doesn't need to actively participate in real time with the scheduler and know things like queue lengths, that was kind of the point for going with the "weight based" approach.

> in fact, this discussion shows that this is (a) an expertly to tune

I'm not so sure that's true- because "tuning" the value would always require experimentation to find the point at which workers are no longer bottlenecked on not having enough splits.

> (b) tuning this is a problem that blurs engine/connector separation. Things may need to be configured on both sides for optimal behavior.

I agree that the implementation is extending the connector's ability to influence the split scheduler, but that's by design. The risk of being misconfigured, however- is fairly small. There's a wide region of "harmless suboptimality" where a better weight assignment scheme either a) split weights are too high and performance could be better if workers didn't spend idle cycles waiting for the next batch of splits to be assigned or b) split weights are too low and the split queues are deeper than they need to be to stay busy. In practice, the "catastrophicly misconfigured" opportunity requires a connector to be extremely aggressive in the weight assignments, which would be a bug that connectors should be expected to fix and can be mitigated by disabling weighted scheduling within that connector if need be.